### PR TITLE
Add wait() to deploy txs

### PR DESCRIPTION
**Pull request by samajammin**
Refactor deploy.ts script to properly wait for txs to execute.

Previously we were calling `await` on deploy txs (e.g. `const tx = await contract.deploy()` but that merely waits for promise to resolve for the tx receipt. In order to wait for the tx to execute / be confirmed, we need to add `await tx.deployTransaction.wait()`.

### CODE REVIEW

**Comment by samajammin** [comment]
@corwintines take a look here as a starting point. Curious to see if this works for if deploying to Rinkeby.

**Comment by samajammin** [comment]
Nice! I was able to run my refactored script & successfully deploy to Rinkeby: https://rinkeby.etherscan.io/address/0x3933775402a2e7316Fc4d02169b72b91f9D5Cc1d

File: contracts/scripts/deploy.ts [hunk 1]
```diff
@@ -15,8 +16,12 @@ async function main() {
...
     maciFactory.address
   )
-  await fundingRoundFactory.deployed()
-  await maciFactory.transferOwnership(fundingRoundFactory.address)
+  await fundingRoundFactory.deployTransaction.wait()
+
+  const transferReceipt = await maciFactory.transferOwnership(
+    fundingRoundFactory.address
+  )
+  await transferReceipt.wait()
```
Comment (samajammin) [inline contracts/scripts/deploy.ts]: TBH I'm not sure if this one is necessary. Not sure if we just need to `wait()` on deploy txs or if we need to `wait()` on every tx.

**Comment by corwintines** [comment]
Added wait to transaction requests that were needed to keep in sync for Rinkeby. This will be useful for local development, but added a new script in https://github.com/ethereum/clrfund/pull/223 to have a more fresh instance without users and projects being created.

**Comment by samajammin** [comment]
FYI @corwintines - I added some missing `wait()`s on the deployTestRound.ts script - please review when able. Also tried to simplify some duplicate code with a few loops.

**Review by corwintines — APPROVED** [review]

**Pull request merged**